test/integration/endpoint_admission_test.go

```diff
@@ -0,0 +1,125 @@
+// +build integration
+
+package integration
+
+import (
+	"testing"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/restclient"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+
+	"github.com/openshift/origin/pkg/client"
+	configapi "github.com/openshift/origin/pkg/cmd/server/api"
+	"github.com/openshift/origin/pkg/cmd/server/bootstrappolicy"
+	testutil "github.com/openshift/origin/test/util"
+	testserver "github.com/openshift/origin/test/util/server"
+)
+
+const (
+	clusterNetworkCIDR = "10.128.0.0/14"
+	serviceNetworkCIDR = "172.30.0.0/16"
+)
+
+var exampleAddresses = map[string]string {
+	"cluster":  "10.128.0.2",
+	"service":  "172.30.0.2",
+	"external": "1.2.3.4",
+}
+
+func setupTestEndpointAdmission(t *testing.T) (kubeClient *kclient.Client, client *client.Client, clientConfig *restclient.Config) {
+	testutil.RequireEtcd(t)
+	masterConfig, err := testserver.DefaultMasterOptions()
+	if err != nil {
+		t.Fatalf("error creating config: %v", err)
+	}
+	masterConfig.KubernetesMasterConfig.AdmissionConfig.PluginConfig = map[string]configapi.AdmissionPluginConfig{
+		"RestrictedEndpointsAdmission": {
+			Configuration: &configapi.DefaultAdmissionConfig{},
+		},
+	}
+	masterConfig.NetworkConfig.ClusterNetworkCIDR = clusterNetworkCIDR
+	masterConfig.NetworkConfig.ServiceNetworkCIDR = serviceNetworkCIDR
+
+	var kubeConfigFile string
+	kubeConfigFile, err = testserver.StartConfiguredMaster(masterConfig)
+	if err != nil {
+		t.Fatalf("error starting server: %v", err)
+	}
+	kubeClient, err = testutil.GetClusterAdminKubeClient(kubeConfigFile)
+	if err != nil {
+		t.Fatalf("error getting kube client: %v", err)
+	}
+	client, err = testutil.GetClusterAdminClient(kubeConfigFile)
+	if err != nil {
+		t.Fatalf("error getting client: %v", err)
+	}
+	clientConfig, err = testutil.GetClusterAdminClientConfig(kubeConfigFile)
+	if err != nil {
+		t.Fatalf("error getting client config: %v", err)
+	}
+	return
+}
+
+func testEndpointAdmission(t *testing.T, client *kclient.Client, namespace, addrType string, success bool) {
+	testEndpoint := &kapi.Endpoints{}
+	testEndpoint.GenerateName = "test"
+	testEndpoint.Subsets = []kapi.EndpointSubset{
+		{
+			Addresses: []kapi.EndpointAddress{
+				{
+					IP: exampleAddresses[addrType],
+				},
+			},
+			Ports:     []kapi.EndpointPort{
+				{
+					Port:     9999,
+					Protocol: kapi.ProtocolTCP,
+				},
+			},
+		},
+	}
+
+	_, err := client.Endpoints(namespace).Create(testEndpoint)
+	if err != nil && success {
+		t.Fatalf("unexpected error creating %s network endpoint: %v", addrType, err)
+	} else if err == nil && !success {
+		t.Fatalf("unexpected success creating %s network endpoint", addrType)
+	}
+}
+
+func TestEndpointAdmissionClusterAdmin(t *testing.T) {
```

Comment:
test setup is a little expensive (~7-10 seconds per test on Jenkins), so I would unify the test cases and do this as a table test instead